File: js/BinaryTree/Node.js
class Node {
  constructor (parent, data) {
    this.data = data
    this.children = []
    if (!parent || parent.constructor !== Node) return
    parent.addChild(this)
  }

  addChild (node) {
    this.children.push(node)
  }
}

module.exports = Node

File: py/binarytree/node.py
class Node(object):
    def __init__(self, parent, data):
        self.data = data
        self.children = []
        if not parent or not isinstance(self, Node):
            return
        parent.add_child(self)

    def add_child(self, node):
        self.children.append(node)
